src/Components/TrackAudioForm/TrackAudioForm.tsx

```diff
@@ -24,30 +25,33 @@ const TrackAudioForm = ({ onFormSubmission, trackData }: TrackFormProps) => {
   const watchedFile = watch('audioFile');
 
   const isModified =
-    (trackData.audioFile && watchedFile === '') ||
+    (trackData.audioFile && watchedFile === null) ||
     (!trackData.audioFile && watchedFile instanceof File) ||
     (trackData.audioFile && watchedFile instanceof File) ||
     false;
 
-  const onSubmit = (data: AudioData) => {
-    const formData = {
-      audioFile: data?.audioFile ?? '',
-    };
-
-    if (formData.audioFile) {
-      uploadAudioTrack(formData.audioFile)
-        .then((data) => handleAddAudioTrack(trackData.id, data.audioFile))
-        .then(() => onFormSubmission());
-    } else {
-      deleteAudioFile()
-        .then(() => handleDeleteAudioTrack(trackData.id))
-        .then(() => onFormSubmission());
+  const onSubmit = async (data: AudioData) => {
+    try {
+      const audioFile = data.audioFile ?? null;
+
+      if (audioFile) {
+        const uploaded = await uploadAudioTrack(audioFile);
+        handleAddAudioTrack(trackData.id, uploaded.audioFile);
+      } else {
+        await deleteAudioFile();
+        handleDeleteAudioTrack(trackData.id);
+      }
+
+      onFormSubmission();
+    } catch (error: unknown) {
+      const err = error instanceof Error ? error : new Error('Something went wrong');
+      toast.error(`Error! ${err.message}`);
     }
   };
 
   return (
     <Form {...form}>
-      <form onSubmit={handleSubmit(onSubmit)} className='space-y-4'>
+      <form onSubmit={(e) => void handleSubmit(onSubmit)(e)} className='space-y-4'>
```

Comment:
It looks a bit overwhelmed and hard to read.
It is better to use wrapper like
```
  const handleFormSubmit = handleSubmit((dto) => {
    onSubmit(dto);
  });
```
and then use `onSubmit={handleFormSubmit}` in `<form>`